src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -146,6 +146,8 @@ private TestPackage CreateTestPackage(string assemblyName)
                     package.Settings[PackageSettings.NumberOfTestWorkers] = workers;
             }
 
+            package.Settings[PackageSettings.SynchronousEvents] = Settings.SynchronousEvents;
```

Comment:
I'd test Settings.InProcDataCollectorsAvailable here and set all three package values, giving a warning if the user has provided any value for NumberOfTestWorkers other than 0 - i.e. it must be either zero or not set.